src/reanimated2/animation/decay.ts

```diff
@@ -76,14 +77,11 @@ export function withDecay(
 
         const deltaTime = Math.min(now - lastTimestamp, 64);
         const clampIndex = initialVelocity > 0 ? 1 : 0;
-        let derivative = 0;
-        if (current < config.clamp![0] || current > config.clamp![1]) {
-          derivative = current - config.clamp![clampIndex];
-        }
+        const derivative = current - config.clamp![clampIndex];
 
-        if (derivative !== 0) {
+        if (Math.abs(derivative) > EPSILON) {
           animation.springActive = true;
-        } else if (derivative === 0 && animation.springActive) {
+        } else if (Math.abs(derivative) <= EPSILON && animation.springActive) {
```

Comment:
If the first conditional fails (i.e. `Math.abs(derivative) > EPSILON` is false), this means that `Math.abs(derivative) <= EPSILON` so there's no need to check it again.
```suggestion
        } else if (animation.springActive) {
```